### PR TITLE
New icons and new data-ampdevmode for Validator extension

### LINKED ISSUE #24176: Update Validator Chrome extension to show debug icon instead of error when in dev mode
In #20974 a new `data-ampdevmode` attribute was added which tells the validator to suppress reporting validation errors for elements it appears on, except for the root `html` element:

image

In such case, however, when there is only one validation error and it is due to being in debug mode, it would be very useful if the Validator extension would not show an error icon but rather a debug icon, similar to how it shows a warning icon when there are deprecation issues.

If there are other validation errors other than the one `data-ampdevmode` on the root `html` element, then the Validator extension should continue to show the error icon.

By showing a debug icon, users won't get alarmed when the system is generating markup that is explicitly invalid for dev mode purposes.

Originally suggested in https://github.com/ampproject/amphtml/issues/20974#issuecomment-524403120.

### PR DESCRIPTION
**Pull request by honeybadgerdontcare**
fixes #24176

updates look of existing icons
introduces new dev mode icon when data-ampdevmode is on html tag
removes number of warnings from icon when page is passing validation (e.g. warnings only)
updates background color for failing validation

### CODE REVIEW

File: validator/js/chromeextension/background.js [hunk 1]
```diff
@@ -95,6 +98,20 @@ function getNumberOfWarnings(errors) {
...
 }
 
+/**
+ * Inspects ValidationResults to discover if in Dev Mode.
+ *
+ * @param {!Object<!ValidationResult>} validationResult
+ * @return {boolean}
+ */
+function isInDevMode(validationResult) {
+  if ((validationResult.errors.length == 1) &&
```
Comment (Gregable) [inline validator/js/chromeextension/background.js:108]: I'm not sure this is what you actually want.

data-ampdevmode is enabled on the HTML tag, but doesn't guarantee that there are no additional errors. In fact, enabling it and doing nothing else doesn't suppress any errors, it just adds an error. It might not even be the first error (thought that's likely).

Instead, once the mode is enabled via the `html` tag, if the same attribute is then added to another other tag in the DOM, say `<html data-ampdevmode>...<body data-ampdevmode>...`, then the other tags with this attribute will have their errors, and the errors of their children, suppressed.

This isn't super obvious, but the reason for the feature was that you could hide errors for sections of the document, such as when a wordpress thing injects some js onto the document and wants to detect errors other than the ones from the injected js.

In other words, it's entirely possible that usage of ampdevmode might also come with more than 1 error. Maybe if there are non-devmode errors, you want to show this as non-devmode, I don't know what you actually want. If so, the naming of the fn is misleading and should get some comments to help the future reader.

File: validator/js/chromeextension/background.js [hunk 2]
```diff
@@ -95,6 +98,20 @@ function getNumberOfWarnings(errors) {
...
+      (validationResult.errors[0].code == 'DEV_MODE_ONLY')) {
+    return true;
+  }
+  return false;
```
Comment (Gregable) [inline validator/js/chromeextension/background.js:112]:
```
if (condition) {
  return true;
}
return false;
```

is an antipattern. Just use `return condition;`

File: validator/js/chromeextension/background.js (hunk 1 above)
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js:108]: I was following the request from #24176 which states:

```
In such case, however, when there is only one validation error and it is due to being in debug mode, it would be very useful if the Validator extension would not show an error icon but rather a debug icon, similar to how it shows a warning icon when there are deprecation issues.
```

In my reading of that, the debug icon only shows when `data-ampdevmode` is on `html` and no other errors or any other errors are removed by having `data-ampdevmode` also on those tags.

File: validator/js/chromeextension/background.js (hunk 2 above)
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js]: Done.

File: validator/js/chromeextension/background.js (hunk 1 above)
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js]: Elaborated in comment and included reference to issue 24176.

File: validator/js/chromeextension/background.js [hunk 3]
```diff
@@ -160,7 +190,8 @@ function handleAmpLink(tabId, ampHref) {
 function handleAmpPass(tabId, validationResult) {
   let badgeTitle = '';
   const numWarnings = getNumberOfWarnings(validationResult.errors);
-  if (numWarnings > 0) {badgeTitle = numWarnings.toString();}
+  // No longer display number of warnings
+  // if (numWarnings > 0) {badgeTitle = numWarnings.toString();}
```
[the remark this answers is not in the document]
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js:194]: Can you contact Zewen with that feedback? I'll connect you two via a different channel.

[the remark this answers is not in the document]
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js:194]: I almost did but thought I'd leave it since users are fickle and may want it back. I'll remove it due to to your feedback.

[the remark this answers is not in the document]
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js]: For accessibility besides the specific visual item within the icon, there is also the color and hover text.

@westonruter  Zewen is looking into finessing the icons more. Would you mind if we move forward with this PR for now and update them when the latest icons are available?

[the remark this answers is not in the document]
Comment (westonruter) [reply validator/js/chromeextension/background.js]: Yes, that's fine.

Re: accessibility, color's not enough in itself for users who have color vision deficiencies. We wrestled with that a lot in the WordPress plugin: https://github.com/ampproject/amp-wp/issues/3726

[the remark this answers is not in the document]
Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js]: Agree that in and of themselves these are not enough. Having several ways to differentiate (structure, color, text) makes up for issues in any one specifically.

**Comment by honeybadgerdontcare** [comment]
@Gregable please re-review, requested changes made

File: validator/js/chromeextension/background.js (hunk 1 above)
Comment (Gregable) [reply validator/js/chromeextension/background.js]: Sounds good. I think my interpretation of this was based on the name of the function. The devmode behavior may be enabled in docs where isDevMode returns false, but for the definition you want for the extension, it's doing the right thing.

I'd lightly suggest tweaking the f'n name, but I don't have a strong opinion. Suggestions would be `displayAsDevMode` or `isDevModeOnlyError` or something like that.

Comment (honeybadgerdontcare) [reply validator/js/chromeextension/background.js]: I disagree that isDevMode isn't correct for data-ampdevmode's intentions. If data-ampdevmode is not being used on other tags which are causing errors then it's not really in dev mode. But to move this forward I will rename it to `onlyErrorIsDevMode`.
